packages/charts/react-charts/library/src/components/HorizontalBarChart/HorizontalBarChart.tsx

```diff
@@ -204,6 +210,149 @@ export const HorizontalBarChart: React.FunctionComponent<HorizontalBarChartProps
+      if (value < minSegmentValueDisplayed) {
+        return acc + (minSegmentValueDisplayed - value);
+      }
+      return acc;
+    }, 0);
+
+    // Find the sum of deviations for segments above the minimum value
+    const overMinTotalDeviations = chartData.reduce((acc, point) => {
+      const value = point.horizontalBarChartdata?.x ?? 0;
+      if (value > minSegmentValueDisplayed) {
+        return acc + (value - minSegmentValueDisplayed);
+      }
+      return acc;
+    }, 0);
+
+    // Avg of deviations over the minimum value for all bigger segments
+    const overMinDeviations = chartData
+      .map(point => point.horizontalBarChartdata?.x ?? 0)
+      .filter(value => value > minSegmentValueDisplayed)
+      .map(value => value - minSegmentValueDisplayed);
+
+    const averageOverMinDeviation =
+      overMinDeviations.length > 0
+        ? overMinDeviations.reduce((acc, val) => acc + val, 0) / overMinDeviations.length
+        : 0;
+
+    // if averageOverMinDeviation is less than deviations, we need deal with it later
+    if (averageOverMinDeviation < underMinDeviations) {
+      // Handle the case where the average deviation is less than the total deviations
+    }
+
+    // Scale the portion of the segments larger than the minimum to reclaim space for the segments less
+    // than the minimum and the space needed for the margins
+    let scale = (overMinTotalDeviations - underMinDeviations - totalMarginValue) / overMinTotalDeviations;
+
+    // If the segment count at min. width overflows, fall back to shrinking below the minimum with all segments the same size
+    if (scale < 0) {
+      scale = 0;
+      // Don't let the segments scale below 1 margin worth of space, falling back to allowing horizontal overflow
+      minSegmentValueDisplayed = Math.max(totalBarXValue - totalMarginValue, totalMarginValue * 2) / noOfSegments;
+    }
+
+    // Normalize your chart data such that:
+    // Small segments (value < minSegmentValueDisplayed) are raised to the minimum threshold.
+    // Large segments (value >= minSegmentValueDisplayed) are scaled down using a scale factor:
+
+    const adjustedSegments = chartData.map(point => {
+      const originalValue = point.horizontalBarChartdata?.x ?? 0;
+      // Percent of original
+      const originalPercent = totalBarXValue > 0 ? (originalValue * 100) / totalBarXValue : 0;
+      // Adjust value
+      const adjustedValue =
+        originalValue < minSegmentValueDisplayed
+          ? minSegmentValueDisplayed
+          : (originalValue - minSegmentValueDisplayed) * scale + minSegmentValueDisplayed;
+      // Adjusted percent
+      const adjustedPercent = totalBarXValue > 0 ? (adjustedValue * 100) / totalBarXValue : 0;
+      return {
+        originalValue,
+        originalPercent,
+        adjustedValue,
+        adjustedPercent,
+      };
+    });
+
+    let prevPosition = 0;
+
+    const annotatedBars = adjustedSegments!.map((segment, index) => {
```

Comment:
what happens if there are 19 bars with value of 1 and 1 bar with value of 100. How does the chart look